leaderboard.py

```diff
@@ -0,0 +1,131 @@
+        case 'ANTARCTICA':
+            regionName = 'Antarctica'
+        case 'ASIA':
+            regionName = 'Asia'
+        case 'AUSTRALIA':
+            regionName = 'Australia'
+        case 'EUROPE':
+            regionName = 'Europe'
+        case 'NORTHA':
+            regionName = 'North America'
+        case 'OCEANIA':
+            regionName = 'Oceania'
+        case 'SOUTHA':
+            regionName = 'South America'
+        case _:
+            regionName = ''
+
+  
+    # Limit the number of rows returned
+    limit = 10
+
+
+    if regionName == '':
+        # SQL Query pulls the number of trees planted by region, region id, region name, and centroid
+        # Example: (182490, 6632405, 'Sierra Leone', '{"type":"Point","coordinates":[-11.7927124667898,8.56329593037589]}')
+        sql = '''select r2.*,region.name, ST_AsGeoJSON(centroid) as centroid from (
+        select count(r.region_id) as planted,r.region_id as region_id from (
+        select distinct(tree_id),region_id  from public.active_tree_region
+        left join region_type on type_id = region_type.id
+        where region_type.type = 'country') r
+        group by region_id
+        order by planted desc
+        limit {}) r2
+        left join region
+        on r2.region_id = region.id'''.format(limit)
+    else:
+        sql = '''
+        select r.*, region.name, ST_AsGeoJSON(region.centroid) as centroid  from (
+        select count(region.id) as planted, region.id
+        from (
+            select trees.*  from trees, region c
+            where c.name = '{}' and ST_WITHIN(trees.estimated_geometric_location, c.geom)
+        ) as trees_in_continent
+        LEFT JOIN region
+        on ST_WITHIN(trees_in_continent.estimated_geometric_location, region.geom)
+        left join region_type
+        on region.type_id = region_type.id
+        where
+        region_type.type = 'country'
+        group by region.id
+        order by count(region.id) desc
+        limit {}
+        ) r left join region
+        on r.id = region.id'''.format(regionName,limit)
+
+    try:
+        # Setup the cursor to execute select statements
+        cur = conn.cursor()
+
+        # Execute the SQL statement
+        cur.execute(sql)
+    
+        # Get the first country
+        row = cur.fetchone()
+
+    except (Exception, psycopg2.DatabaseError) as error:
+        print(error)
+
+
+    # List of dictionary objects to return
+    leading_countries = []
+
+    # Dictionary key values
+    keys = ['planted', 'id', 'name', 'centriod']
+
+    # Dictionary object that holds the country's information using keys above
+    country = {}
+
+    # Create the dictionary object and append it to a list
+    while row is not None:
+        index = 0
+        for column in row:
+            country[keys[index]] = column
+            index += 1
+        leading_countries.append(country.copy())
+        row = cur.fetchone()
+
+    for item in leading_countries:
```

Comment:
@CareyAltgilbers did you insert/update the leader board into the config table in the DB?

Reply:
@dadiorchen Thank you for letting me know how to connect to the database. I did some research and looked at your code and think I know how to setup the airflow to postgreSQL connection. I'm going to work on that tomorrow. Thank you for pointing me in the right direction!

Reply:
@dadiorchen  For this submission I removed the database.ini file and the references in the leaderboard.py file to that connection and replaced it with the following connection syntax:

from airflow.providers.postgres.operators.postgres import PostgresOperator
from airflow.providers.postgres.hooks.postgres import PostgresHook
postgresConnId = "postgres_default"
db = PostgresHook(postgres_conn_id=postgresConnId)
conn = db.get_conn()

I am not able to test to see if this works from my test environment.  It would be great if you would be able to test and let me know if the leaderboard.py file is returning a list of leading country dictionary objects.

The pre_populate_leader_boarder.py file is my attempt at the airflow DAG. I am learning airflow now and was not sure yet how to just add the python function in the DAG so I created a DAG file to call the 
leaderboard.py. 

When the leaderboard.py file is called from the command line we can pass an argument for the region that will hopefully work with the second SQL statement in the file once it is rewritten by Mohmim. 

I am not sure how to pass a variable into the DAG file yet to set the region variable.

Reply:
@CareyAltgilbers I think it is incorrect to connect DB this way, and you didn't insert the data to webmap.config table, right?

Reply:
That is correct, I am not inserting the data into the web map.config table. I was not sure where to send the result of the function. I'll setup a Google Meet so we can go over it. Thank you.